conda/core/solve.py

```diff
@@ -361,9 +370,7 @@ def _find_inconsistent_packages(self, ssc):
             The following packages are causing the inconsistency:"""))
             print(dashlist(inconsistent_precs))
             for prec in inconsistent_precs:
-                # pop and save matching spec in specs_map
-                spec = ssc.specs_map.pop(prec.name, None)
-                ssc.add_back_map[prec.name] = (prec, spec)
+                ssc.add_back_map[prec.name] = (prec, None)
```

Comment:
Why does this have a tuple of `(prec, None)` opposed to just the prec?

Reply:
That looks like a bug to me.  The spec is otherwise not removed from the specs_map, which is probably important.

Reply:
This one is more complicated than I thought.  Here's how it works:

With pop:
- inconsistent dep is removed
- none of its specs become part of the solution (and may thus be eliminated)
- package is restored, and its deps may or may not be there

Without pop:
- inconsistent dep is not factored into the solution, but it is entered in as a name spec
- solution goes ahead with any dependencies satisfied
- consistent package will get clobbered by the original inconsistent spec

I'll push up some changes here.